src/components/ChangeOracles.tsx

```diff
@@ -0,0 +1,286 @@
+import { Menu, Transition } from '@headlessui/react';
+import clsx from 'clsx';
+import { AnimatePresence, motion } from 'framer-motion';
+import { LoadStatus, usePublishedDatum, WalletContext } from 'lib/wallet';
+import {
+  Fragment,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { FiChevronDown, FiPlus, FiX } from 'react-icons/fi';
+import { bech32 } from 'bech32';
+
+interface ListItemProps {
+  address: string;
+  onRemove: () => void;
+}
+
+const ListItem = ({ address, onRemove }: ListItemProps) => {
+  return (
+    <div className="flex flex-row justify-between  text-purple-500 bg-purple-50 rounded-lg p-2">
+      <span className="leading-8">{address}</span>
+      <button
+        className="p-2 rounded hover:bg-purple-200"
+        type="button"
+        onClick={onRemove}
+      >
+        <FiX />
+      </button>
+    </div>
+  );
+};
+
+export enum ChangeOraclesMode {
+  Add = 0,
+  Remove = 1,
+}
+
+interface Props {
+  charterOfferId: number;
+  mode: ChangeOraclesMode;
+}
+
+export default function ChangeOracles({ charterOfferId, mode }: Props) {
+  const walletUtils = useContext(WalletContext);
+  const addressInput = useRef(null);
+
+  const { data: instances, status } = usePublishedDatum('agoricNames.instance');
+  const priceFeeds = (
+    status === LoadStatus.Received
+      ? instances.filter(([name]) => name.includes('price feed'))
+      : []
+  ).map(([name]) => name);
+
+  const [priceFeed, setPriceFeed] = useState(null);
+  const [addresses, setAddresses] = useState([]);
+  const [currentInput, setCurrentInput] = useState('');
+  const [minutesUntilClose, setMinutesUntilClose] = useState(10);
+
+  const inputError = useMemo(() => {
+    if (!currentInput) {
+      return '';
+    }
+
+    if (addresses.find(a => a === currentInput)) {
+      return 'Already added';
+    }
+    try {
+      const { prefix } = bech32.decode(currentInput);
+      return prefix === 'agoric' ? '' : 'Invalid address';
+    } catch (e) {
+      return 'Invalid address';
+    }
+  }, [addresses, currentInput]);
+
+  const removeAddress = useCallback(
+    address => {
+      setAddresses(addresses.filter(a => a !== address));
+    },
+    [addresses],
+  );
+
+  const addAddress = useCallback(() => {
+    setAddresses([...addresses, currentInput]);
+    setCurrentInput('');
+    addressInput.current.focus();
+  }, [addresses, currentInput]);
+
+  const canMakeProposal = !!charterOfferId && addresses.length > 0;
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    console.debug({ event, addresses, minutesUntilClose });
+    const offerFn = (() => {
+      switch (mode) {
```

Comment:
happy to see exhaustive switch knowing `default` unnecessary.

though in this case I think an object map would be simpler? consider,
```
const offerFns = {
  ChangeOraclesMode.Add: walletUtils.makeVoteOnAddOracles,
  ChangeOraclesMode.Remove: walletUtils.makeVoteOnRemoveOracles,
}
const offer = offerFn[mode](
```